v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -174,10 +176,9 @@ static SourceSchema getSourceSchema(
    */
   private static ImmutableList<TableConfig> autoInferTableConfigs(
       JdbcIOWrapperConfig config, SchemaDiscovery schemaDiscovery, DataSource dataSource) {
-    ImmutableList<String> tables =
-        (config.tables().isEmpty())
-            ? schemaDiscovery.discoverTables(dataSource, config.sourceSchemaReference())
-            : config.tables();
+    ImmutableList<String> discoveredTables =
+        schemaDiscovery.discoverTables(dataSource, config.sourceSchemaReference());
+    ImmutableList<String> tables = getTablesToMigrate(config.tables(), discoveredTables);
```

Comment:
This can probably be done earlier, and if the list is empty - then no need to even discover source schema and fail/stop fast. But just a micro-optimization, won't press on this.

Reply:
If the list is empty, we need to fallback to migrating all the possible tables.